pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+"""
+Defines parameter sets used to set the behavior for core pypit
+functionality.
+
+There are two main parameter set groups, those used for reducing the
+data:
+
+    - RunPar: Parameters specific to a given execution of PypIt
+
+    - ReducePar: Parameters the define how PypIt should perform the
+      reductions.  These include general parameters and the following
+      parameter subsets:
+        - OverscanPar: Methods used for the overscan subtraction.
+        - FlatFieldPar: Methods used for field-flattening.
+        - SkySubtractionPar: Methods used for sky subtraction.
+        - FlexurePar: Methods used for flexure correction.
+        - WavelengthCalibrationPar: Methods used for constructing the
+          wavelength solution.
+        - FluxCalibrationPar: Methods used for flux calibration.
+
+    - FrameGroupPar: Sets parameters that are used to group and combined
+      frames.
+        - CombineFramesPar - Parameters for combining frames
+
+    - WavelengthSolutionPar: Parameters used for constructing the wavelength
+      solution
+
+    - TraceSlitPar: Parameters for tracing slits
+        - PCAPar: PCA parameters
+
+    - TraceTiltsPar: Parameters for tracing tilts of arc lines
+
+    - TraceObjectsPar: Parameter for tracing objects
+
+    - ExtractObjectPar: Parameters for extracting 1D object spectra
+        - ManualExtractionPar: Parameters needed for manual extraction
+          of 1D spectra
+
+And those that define the parameters of the instrument used to collect
+the data:
+
+    - DetectorPar: Specifies the properties of a given detector
+
+    - InstrumentPar: Specifies the instrument used for a given
+      observation and contains the list of detectors for the camera.
+
+    - FrameFitsPar: Defines the set of fits files properties that all
+      fits files from the given instrument should have.
+
+    - FrameIDPar: Defines a set of parameters used to identify fits
+      files as a certain frame type.
+
+These are collected into the main, high-level paramter set, called
+PypitPar.
+
+.. todo::
+    - Warn the user if a default value is being used?
+
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import sys
+import os
+import glob
+import warnings
+import numpy
+import inspect
+
+from configobj import ConfigObj
+from astropy.time import Time
+
+from .parset import ParSet
+
+#-----------------------------------------------------------------------------
+# Helper functions
+
+def _pypit_root_directory():
+    """
+    Determine the root directory for the PYPIT source code.
+
+    .. todo::
+        - Set this in __init__.py
+    """
+    for p in sys.path:
```

Comment:
pretty sure this works without fail:

```
from pkg_resources import resource_filename
p = resource_filename('pypit', '')
```

Reply:
Thanks!  I've changed the code to use this instead.